### PR TITLE
Misc discovery and execution cache tweaks

### PR DESCRIPTION
**Pull request by mattgodbolt**
- use async file access when looking for llc
- don't use the random customCwd in the cache hash
- don't lookup "log cache accesses" in props every access
- turn on compiler cache access logging in discovery mode
- error on missing wine executable config
- atomically cache on disk file cache

### CODE REVIEW

File: lib/buildenvsetup/base.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export class BuildEnvSetupBase {
         } else if (this.compilerTypeOrGCC === 'clang') {
             const binpath = path.dirname(this.compiler.exe);
             const llc = path.join(binpath, 'llc');
-            if (fs.existsSync(llc)) {
+            if (await fs.exists(llc)) {
```
Comment (partouf) [inline lib/buildenvsetup/base.ts:87]: fs.exists is officially deprecated, you can use await utils.fileexists though

Comment (mattgodbolt) [reply lib/buildenvsetup/base.ts:87]: ta!

File: lib/base-compiler.ts [hunk 2]
```diff
@@ -378,12 +378,16 @@ export class BaseCompiler implements ICompiler {
...
+        const optionsForCache = {...options};
         if (options.createAndUseTempDir) {
-            const tmpDir = await this.newTempDir();
-            options.customCwd = tmpDir;
+            options.customCwd = await this.newTempDir();
         }
 
-        const key = this.getCompilerCacheKey(compiler, args, options);
+        const key = this.getCompilerCacheKey(compiler, args, optionsForCache);
+        if (options.createAndUseTempDir) delete key.options.customCwd;
```
Comment (partouf) [inline lib/base-compiler.ts:390]: Isn’t this redundant? Its already using the key before options was extended with customcwd

Comment (mattgodbolt) [reply lib/base-compiler.ts:390]: Oh shucks, good spot. Yeah it absolutely is and in fact is _the point_ of me taking a copy before....thanks!

**Review by partouf — APPROVED** [review]

**Pull request merged**